Userland/Utilities/locate.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2021, Aron Lander <aron@aronlander.se>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <LibCore/LocateDb.h>
+#include <LibCore/ArgsParser.h>
+#include <AK/String.h>
+
+int main(int argc, char** argv) {
+
+    Vector<const char*> keyword_holder;
+    bool o_case_insensitive = false;
+
+    Core::ArgsParser args_parser;
+    args_parser.add_option(o_case_insensitive, "Ignore case during keyword matching", "ignore-case", 'i');
+    args_parser.add_positional_argument(keyword_holder, "Keyword to search for within the file db", "keyword", Core::ArgsParser::Required::Yes);
+    args_parser.parse(argc, argv);
+
+    auto locate_db = Core::LocateDb("/home/anon/testdb.db", Core::LocateDbMode::read);
+    VERIFY(locate_db.verify_header());
+    String keyword = String(keyword_holder[0]);
+    CaseSensitivity case_sensitivity = o_case_insensitive ? CaseSensitivity::CaseInsensitive : CaseSensitivity::CaseSensitive;
+
+    while (auto directory = locate_db.get_next_directory()) {
+        if (directory->path.contains(keyword, case_sensitivity)) {
+            printf("%s\n", directory->path.characters());
```

Comment:
You could also use `outln("{}", directory->path);` if you want.
one less `strlen` :shrug: